rules/prefer-string-slice.js

```diff
@@ -41,10 +74,46 @@ const create = context => {
 				message: 'Prefer `String#slice()` over `String#substring()`.'
 			};
 
-			const canFix = argumentNodes.length === 0;
+			const firstArgument = argumentNodes[0] ? sourceCode.getText(argumentNodes[0]) : undefined;
+			const secondArgument = argumentNodes[1] ? sourceCode.getText(argumentNodes[1]) : undefined;
+
+			const firstNumber = argumentNodes[0] ? getNumericValue(argumentNodes[0]) : undefined;
+
+			let slice;
+
+			if (argumentNodes.length === 0) {
+				slice = '.slice()';
+			} else if (argumentNodes.length === 1 && firstNumber !== undefined) {
+				slice = `.slice(${Math.max(0, firstNumber)})`;
+			} else if (argumentNodes.length === 1 && isLengthProperty(argumentNodes[0])) {
+				slice = `.slice(${firstArgument})`;
+			} else if (argumentNodes.length === 1) {
```

Comment:
merge this logic in one `argumentNodes.length === 1` block should be more readable